tests/resources_t/test_drone.py

```diff
@@ -100,9 +100,10 @@ async def mocked_run(drone):
         with self.assertLogs(level=DEBUG):
             run_async(self.drone.run)
 
-        mocked_asyncio_sleep.assert_called_once_with(
-            self.mock_site_agent.drone_heartbeat_interval
-        )
+        # duration skipped via asyncio.sleep
+        # use the `sum` to avoid `asyncio.sleep(0)` context switches to skew the result
+        mock_elapsed = sum(args[0] for args, _ in mocked_asyncio_sleep.call_args_list)
+        self.assertEqual(mock_elapsed, self.mock_site_agent.drone_heartbeat_interval)
```

Comment:
I don't think my changes actually impact what is being tested here. Running the testsuite locally, the initial test sporadically failed – this looks like there is some unstable order dependency, e.g. on the order in ``set``s.

I'm not sure whether the test being order dependent is the actual problem, or whether it hints at another issue.

Reply:
Hmm, I have never experienced this issue before. Do you remember what the error message was in your local tests?

Reply:
The error message from the test is in [this comment](https://github.com/MatterMiners/tardis/pull/224#issuecomment-1004072529). (Had to search it myself – the commits and LGTM comments sure are a lot of noise.)